### PR TITLE
Instead of erroring, clean up after dangling IDs in DB

### PR DESCRIPTION
**Pull request by mheon**
For various (mostly legacy) reasons, Podman presently maintains a unified namespace for pods and containers - IE, we cannot have both a pod and a container named "test" at the same time. To implement this, we use a global database table of every pod and container ID (and another of every pod and container name).

These entries should be added when containers/pods are added, and removed when containers/pods are removed, with the database's transactional integrity providing a guarantee that this is batched with the overall removal and that the DB should remain sane and consistent no matter what. As such, we treat a dangling ID as a hard error that stops the use of Podman.

Unfortunately, we have someone run into this last Friday. I'm still not certain how exactly their DB got into this state, but without further clarification there, we can consider removing the error and making Podman instead clean up and remove any dangling
IDs, which should restore Podman to a serviceable state. Drop an error message if we do this, though, because people should know that the DB is in a bad state.

[NO NEW TESTS NEEDED] it is deliberately impossible to produce a configuration that would test this without hex-editing the DB file.

```release-note
Fixed a bug where a dangling ID in the database could render Podman unusable.
```

### CODE REVIEW

**Review by vrothberg — APPROVED** [review]
LGTM

File: libpod/boltdb_state.go [hunk 1]
```diff
@@ -192,15 +197,23 @@ func (s *BoltState) Refresh() error {
...
-					return errors.Wrapf(define.ErrInternal, "id %s is not a pod or a container", string(id))
+					// Something is seriously wrong, but
+					// continue on and try to clean up the
+					// state and become consistent.
+					// Just note what needs to be removed
+					// for now - ForEach says you shouldn't
+					// remove things from the table during
+					// it.
+					logrus.Errorf("Database issue: dangling ID %s found (not a pod or container) - removing", string(id))
+					toRemoveIDs = append(toRemoveIDs, string(id))
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply libpod/boltdb_state.go:216]: very good catch! I agree.

[the remark this answers is not in the document]
Comment (mheon) [reply libpod/boltdb_state.go:216]: Fixed

[the remark this answers is not in the document]
Comment (Luap99) [reply libpod/boltdb_state.go:216]: oh sry this is not an actual for loop, just a function so you have to return

**Comment by mheon** [comment]
Bot's broken, doing it manual-like

**Pull request merged**